pkg/openshift-sdn/proxy.go

```diff
@@ -45,16 +44,13 @@ func readProxyConfig(filename string) (*kubeproxyconfig.KubeProxyConfiguration,
 	if err := o.Complete(); err != nil {
 		return nil, err
 	}
-
 	config := o.GetConfig()
-
-	// o.Complete() will set the feature gates from the config, but we need to re-set
-	// them with EndpointSlice forced off
-	config.FeatureGates[string(features.EndpointSlice)] = false
-	config.FeatureGates[string(features.EndpointSliceProxying)] = false
-	utilfeature.DefaultMutableFeatureGate.SetFromMap(config.FeatureGates)
-
-	return config, nil
+	if config.FeatureGates[string(features.EndpointSlice)] ||
+		config.FeatureGates[string(features.EndpointSliceProxying)] {
+		return nil, fmt.Errorf("Unsupported feature gate(s) enabled in proxy config. Feature gates: %v",
+			config.FeatureGates)
+	}
```

Comment:
hm... I was thinking more that `runProxy` would check `if utilfeature.DefaultFeatureGate.Enabled(features.EndpointSliceProxying)` and bail out there.

(That will also protect us by making things fail immediately when the feature gate becomes on-by-default.)

Reply:
the `readProxyConfig` is happening in `ValidateAndParse()` before `Start()` though. Don't we want to fail fast?

Reply:
It's still at startup time. The point is that we want to fail "if EndpointSliceProxing is enabled", not "if EndpointSliceProxying was enabled via command-line options"

Reply:
Are there other paths besides command line args by which the gate could be enabled?

Reply:
I guess I see your point. But I was hoping we will actually add support for EndpointSlices in o-sdn by the time the FeatureGate is turned on-by-default.